### DISCUSSION TITLE
GitHub Actions Status Line

### DISCUSSION THREAD

**Discussion by thetayloredman** [community/community#24013] [New to GitHub]
<p>Sometimes, I see workflows that produce a little status line.</p>
<p>For example, the CodeQL workflow “duplicates” it’s status entry and shows a little bit of output:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/0/301033a14bf33dcddf2242f4dd3dd5f41052eb46.png" data-download-href="/uploads/short-url/6RbD5IKjH8kpQTZpLVilIFQlufQ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/26350849/181080634-5adab745-bbe4-468a-91a0-4a3c4d50521e.png" alt="image" data-base62-sha1="6RbD5IKjH8kpQTZpLVilIFQlufQ" width="690" height="62" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/0/301033a14bf33dcddf2242f4dd3dd5f41052eb46_2_690x62.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/0/301033a14bf33dcddf2242f4dd3dd5f41052eb46_2_1035x93.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/0/301033a14bf33dcddf2242f4dd3dd5f41052eb46.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/0/301033a14bf33dcddf2242f4dd3dd5f41052eb46_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1234×112 11.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is it possible for me to do this myself? It would be cool if I could set my workflow up so it said something like “Lint errors found: src/index.ts line 5: <span class="mention">@typescript-eslint</span>/no-explicit-any, … and 5 more”</p>

**Comment by aeisenberg** [comment, marked as answer]
<p>This is done through the <a href="https://docs.github.com/en/rest/reference/repos#statuses">GitHub status API</a>. I don’t <em>think</em> you can update the status of a workflow, since that is handled directly by GitHub Actions (but try it…I might be wrong).</p>
<p>Instead, you can create your own status by using the GitHub CLI like this in the workflow:</p>
<pre><code class="lang-auto">gh api -X POST "repos/$REPOSITORY/statuses/$SHA" \
            -f state="failure" \
            -f description="Lint errors found: src/index.ts line 5: @typescript-eslint/no-explicit-any" \
            -f context="My tslinting"
</code></pre>
<p>Read the full docs for all the details.</p>
